lib/dockerregistry/inventory.go

```diff
@@ -784,68 +784,69 @@ func ToPQIN(registryName RegistryName, imageName ImageName, tag Tag) string {
 	return string(registryName) + "/" + string(imageName) + ":" + string(tag)
 }
 
-func getImagesSliceFromMap(images map[ImageName]DigestTags) []Image {
-	var result []Image
+// ToYAML displays a RegInvImage as YAML, but with the map items sorted
+// alphabetically.
+func (rii *RegInvImage) ToYAML() string {
+	// Temporary structs that have slices, not maps.
+	type digest struct {
+		hash string
+		tags []string
+	}
 
-	for name, dmap := range images {
-		result = append(result, Image{
-			ImageName: name,
-			Dmap:      dmap,
-		})
+	type image struct {
+		name    string
+		digests []digest
 	}
 
-	return result
-}
+	var images []image
+
+	for name, dmap := range *rii {
+		var digests []digest
+		for k, v := range dmap {
+			var tags []string
+			for _, tag := range v {
+				tags = append(tags, string(tag))
+			}
 
-// getImagesFromDestRegistry returns all images from not src repository
-func (sc *SyncContext) getImagesFromDestRegistry(
-	srcRegistryName RegistryName,
-) ([]Image, error) {
-	var registries []RegInvImage
+			sort.Strings(tags)
 
-	for registryName, images := range sc.Inv {
-		// Ignore src registry
-		if registryName == srcRegistryName {
-			continue
+			digests = append(digests, digest{
+				hash: string(k),
+				tags: tags,
+			})
 		}
+		sort.Slice(digests, func(i, j int) bool {
+			return digests[i].hash < digests[j].hash
+		})
 
-		registries = append(registries, images)
-	}
-
-	if len(registries) != 1 {
-		return nil, fmt.Errorf("more than 1 destination repository" +
-			" is currently not supported")
+		images = append(images, image{
+			name:    string(name),
+			digests: digests,
+		})
 	}
 
-	return getImagesSliceFromMap(registries[0]), nil
-}
+	sort.Slice(images, func(i, j int) bool {
+		return images[i].name < images[j].name
+	})
 
-func getSrcRegistryFromManifest(manifest Manifest) RegistryContext {
-	var srcRegistry RegistryContext
-	for _, registry := range manifest.Registries {
-		if registry.Src {
-			srcRegistry = registry
+	var b strings.Builder
```

Comment:
Why not just use `yaml.Marshal` (from `sigs.k8s.io/yaml`)?

Reply:
Or ... can we make this format match the format of the manifest?  This could also be super-handy for creating the manifest, I think!

Reply:
> Why not just use `yaml.Marshal` (from `sigs.k8s.io/yaml`)?

This is because we want to deterministically (alphabetically) sort the map entries (these are the items under `dmap`, as well as the images in the list. I did some trial runs with `yaml.Marshal` and the items were not listed in a sorted fashion --- hence this scheme.

I could use yaml.MapSlice{} or something but this was simpler.

That said... does `sigs.k8s.io/yaml` already do this kind of sorting automatically? But at the same time I do want `name:` to come before `dmap:`, so... I want *mostly* alphabetical sorting of the key elements.

Reply:
> Or ... can we make this format match the format of the manifest? This could also be super-handy for creating the manifest, I think!

It already matches the format of a subset of the promoter manifest (`images` key portion). We can use it later to build an complete manifest dynamically, I suppose.

Reply:
Ah - ordering makes sense, and I love that it matches the input format.  We should put that in a tip somewhere as I don't want to tell you how I was building the list when I've been doing this...